snap/local/README.md

```diff
@@ -0,0 +1,411 @@
+
+To configure the DU Agent Module principal file (.toml) in Ubuntu jCore, you need to create the file at /etc/aziot/identityd/snap-aziot-du.toml. The file should contain the following information:
+
+```toml
+[[principal]]
+name = "IoTHubDeviceUpdate"
+idtype = ["module"]
+uid = <SNAP_AZIOT_DU_USER_ID>
+```
+
+Replace `SNAP_AZIOT_DU_USER_ID` with the actual user id of `snap_aziot_du`
+
+This specifies the principal information for the DU Agent Module, including its name, ID type, and user ID. It is important to ensure that the file is created with the correct filename and location, and that the specified values are appropriate for your specific use case.
+
+
+#### Connect to Other Snap with Specific User ID
+When connecting snaps with interfaces, the snaps are typically connected with the default user or "system" user. However, it is possible to connect snaps with a specific user ID by using the `--classic` and `--username` options with the snap connect command.
+
+For example, to connect to Azure Identity Service with a specific user ID called "snap_aziot_adu", you would use the following command:
+`snap connect --classic --username=snap_aziot_du deviceupdate-agent:AIS-interface azureIdentityService-snap:AIS-interface`  
+This will connect the two snaps using the "snap_aziot_adu" user ID, allowing the snaps to communicate with each other as that user.
+
+It is important to note that using the `--classic` and `--username` options with the snap connect command can have security implications, as it allows the connected snaps to access each other's data and resources as the specified user. Therefore, it should only be used if necessary and with caution.
+
+### Integrating With Another Snaps
+
+The sections below explains how Device Update Agent snap communicate with other snap.
+
+### Delivery Optimization Integration
+
+To download files using DO Agent snap, the DU Agent snap need to connect to the following `slots` provided by the DO Agent snap:
+
+```yaml
+# Note: snippet from https://github.com/microsoft/do-client/blob/develop/snapcraft-options/snapcraft-agent.yaml
+
+slots:
+  do-port-numbers:
+    interface: content
+    content: do-port-numbers
+    read: [ $SNAP_DATA/run ]
+
+  do-configs:
+    interface: content
+    content: do-configs
+    write: [ $SNAP_DATA/etc ]
+
+plugs:
+  deviceupdate-agent-downloads:
+    interface: content
+    content: deviceupdate-agent-downloads
+    target: $SNAP_DATA/deviceupdate-agent-downloads
+
+layout:
+  # DU agent will provide the /var/lib path as download location.
+  # Map it to the correct path within this snap.
+  /var/lib/deviceupdate-agent-downloads:
+    symlink: $SNAP_DATA/deviceupdate-agent-downloads
+```
+
+You can list a DO client snap's connections by using  the `snap connections` command. This command displays information about the snap's connections, including its slots and plugs.
+
+```shell
+adu-dev@du-ubuntu-core-2004-build:~$ snap connections deliveryoptimization-client
```

Comment:
Can remove $PROMPT before the `$` and same for rest of doc such as line 233 to be more consistent like you did on lines 390 and 396

Reply:
Removed